File: DSA Using C/2. Algorithms/Searching Algorithms/binary_search.c
#include <stdio.h>

int binarySearch(int arr[], int n, int key)
{
    int starting_point = 0;
    int ending_point = n;
    while (starting_point <= ending_point)
    {
        int mid = (starting_point + ending_point) / 2;

        if (arr[mid] == key)
        {
            return mid;
        }
        else if (arr[mid] > key)
        {
            ending_point = mid - 1;
        }
        else
        {
            starting_point = mid + 1;
        }
    }
    return -1;
}

int main()
{
    int n;
    printf("Enter size of array:  ");
    scanf("%d", &n);
    int arr[n];
    printf("Enter arrays element in sorting order: \n");
    for (int i = 0; i < n; i++)
    {
        scanf("%d", &arr[i]);
    }
    int key;
    printf("Enter key for searching :   ");
    scanf("%d", &key);
    printf("Element %d presents on index %d", key, binarySearch(arr, n, key));
    return 0;
}

File: DSA Using Java/Algorithms/Searching Algorithm/BinarySearch.java
public class BinarySearch {
    public static int binarySearch(int[] arr, int key){
        int startingPoint = 0;
        int endingPoint = arr.length-1;

        while (startingPoint <= endingPoint){
            int midPoint = (startingPoint + endingPoint)/2;

            if(arr[midPoint] == key){
                return midPoint;
            } else if (arr[midPoint] > key) {
                endingPoint = midPoint - 1;
            } else {
                startingPoint = midPoint + 1;
            }
        }

        return -1;
    }

    public static void main(String[] args) {
        int[] marks = {89, 92, 98, 76, 67, 99, 88, 100, 91, 89};

        int isFound = binarySearch(marks, 99);

        if(isFound == -1){
            System.out.println("Not Found!");
        } else {
            System.out.println("Element Found at Index "+isFound);
        }
    }
}
